modules/userId/index.js

```diff
@@ -605,6 +605,57 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+ * This function will be exposed in global-name-space so that userIds for a source can be exposed
+ * Sample use case is exposing this function to ESP
+ */
+function getEncryptedEidsForSource(source, encrypt, customFunction) {
+  let eids = [];
+  let eidsSignals = {};
+
+  if (typeof customFunction === 'function') {
+    logInfo(`${MODULE_NAME} - Getting encrypted signal from custom function : ` + customFunction.name + `& source : ` + source);
+    // Publishers are expected to define a common function which will be proxy for signal function.
+    var customSignals = customFunction(source);
+    eidsSignals[source] = customSignals ? encryptSignals(customFunction(source)) : null; // by default encrypt using base64 to avoid JSON errors
+  } else {
+    // initialize signal with eids by default
+    eids = (getGlobal()).getUserIdsAsEids();
+    logInfo(`${MODULE_NAME} - Getting encrypted signal for eids : ` + JSON.stringify(eids));
+    eids.forEach((eid) => {
+      eidsSignals[eid.source] = encrypt === true ? encryptSignals(eid) : eid.uids[0].id; // If encryption is enabled append version (1||) and encrypt entire object
+    });
+  }
+  let promise = Promise.resolve(eidsSignals[source]);
+  logInfo(`${MODULE_NAME} - Fetching encrypted eids: ` + eidsSignals[source]);
```

Comment:
We can avoid concatenating using `+`  by using `Template literals`.

Reply:
Done